others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,273 @@
+/**
+ * @file
+ * @brief Iterative version of Preorder, Postorder, and preorder [Traversal of the Tree]
+ * (https://en.wikipedia.org/wiki/Tree_traversal)
+ * @author [Motasim](https://github.com/motasimmakki)
+ * @details
+ * 
+ * ### Iterative Preorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first right child then left child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * And return the result array as the preorder traversal of a tree.
+ * 
+ * ### Iterative Postorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first left child then right child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * Now reverse the result array and then return it to the calling function as a postorder traversal of a tree.
+ * 
+ * ### Iterative Inorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current.
+ * Now iterate and take the current to the extreme left of the tree by traversing only to its left.
+ * Pop the elemnt from the stack and assign it to the current.
+ * Store the data of current into the result array.
+ * Repeat the same set of steps until the Stack becomes empty or the current becomes NULL.
+ * And return the result array as the inorder traversal of a tree.
+ */
+#include <iostream>   /// for I/O operations
+#include <stack>      /// for `stack`
+#include <vector>     /// for `vector`
+#include <algorithm>  /// for `reverse`
+#include <cassert>    /// for `assert`
+
+/**
+ * @namespace others
+ * @brief Other algorithms
+ */
+namespace others {
+/**
+ * @namespace iterative_tree_traversals
+ * @brief Functions for [Traversal of the Tree](https://en.wikipedia.org/wiki/Tree_traversal) algorithm
+ */
+namespace iterative_tree_traversals {
+/**
+ * @brief defines the structure of a node of the tree
+ */
+struct Node {
+    int data = 0;       ///< The value/key of the node.

```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
I modified the data type of `data` from `int` to `int64_t` so that it works correctly with the signed values also.